### PR TITLE
614 ScaleIntensityRangePercentiles transform

### LINKED ISSUE #614: Add percentile version of ScaleIntensityRange
**Describe the solution you'd like**

A variant of `monai.transforms.ScaleIntensityRange` that has upper and lower intensity percentile parameters instead of `a_min` and `a_max`.      

So this transform would rescale each image differently depending on the intensity distribution of the `img` parameter instead of scaling all images the same, as `ScaleIntensityRange` does.

General idea:
```python
class ScaleIntensityRangePercentiles(Transform):
    def __init__(self, lower_percentile: float, upper_percentile: float, b_min: float, b_max: float, clip: bool = False) -> None:
        self.lower_percentile = lower_percentile
        self.upper_percentile = upper_percentile
        self.b_min = b_min
        self.b_max = b_max
        self.clip = clip

    def __call__(self, img):
        ### Calculate percentiles from img
        ### Rescale using percentiles
        ### Optional clipping
```
**Describe alternatives you've considered**
Modify the current `ScaleIntensityRange` transform to include a percentile option:         

`ScaleIntensityRange(a_min, a_max, b_min, b_max, clip=False, percentile=True)`         

Which would treat `a_min` and `a_max` as percentiles instead of intensity values.

### PR DESCRIPTION
**Pull request by abpwrs**
Fixes #614 

### Description
Adds a data transform to rescale an image based on upper and lower intensity percentiles.

### Status
Ready

TODOs:
- [x] Add unit testing
- [x] Add docstring usage 

### Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] Breaking change (fix or new feature that would cause existing functionality to change)
- [x] New tests added to cover the changes
- [x] Docstrings/Documentation updated

### CODE REVIEW

**Comment by Nic-Ma** [comment]
Hi @abpwrs ,

Thanks for your quick PR!

**Comment by hjmjohnson** [comment]
@abpwrs If one chooses 10%, 90%, min=0, max=200,

The current code seems to pin the 10% value to 0 and 90% value to 200. 

===
The algorithm that we have found works best is
define a linear transformation such that 10% maps to the value 20, and the 90% value mapps to the value 180.  Apply that linear transform for all values in the range (0-200) and set values (after passing through the above linear transform) less than 0 to 0, and greater than 200 to 200.

== Perhaps a flag could distinguish between the two variants of the algorithm that people would expect.

Please provide better documentation, with examples of how the transfer function is computed.

**Comment by abpwrs** [comment]
@hjmjohnson I think a flag for this variant sounds great. I'll update this PR accordingly.

File: monai/transforms/intensity/array.py [hunk 1]
```diff
@@ -297,3 +297,89 @@ def __call__(self, img):
...
+        a_min = np.percentile(img, self.lower)
+        a_max = np.percentile(img, self.upper)
+        b_min = self.b_min
+        b_max = self.b_max
+
+        if self.relative:
+            b_min = ((self.b_max - self.b_min) * (self.lower / 100.0)) + self.b_min
+            b_max = ((self.b_max - self.b_min) * (self.upper / 100.0)) + self.b_min
+
+        scalar = ScaleIntensityRange(a_min=a_min, a_max=a_max, b_min=b_min, b_max=b_max, clip=False)
```
Comment (wyli) [inline monai/transforms/intensity/array.py:379]: could you help fix this?
```python
>>> from monai.transforms import ScaleIntensityRangePercentiles
>>> import numpy as np
>>> ScaleIntensityRangePercentiles(1, 99, 0, 1)(np.ones((5, 5)))
/MONAI/monai/transforms/intensity/array.py:239: RuntimeWarning: invalid value encountered in true_divide
  img = (img - self.a_min) / (self.a_max - self.a_min)
array([[nan, nan, nan, nan, nan],
       [nan, nan, nan, nan, nan],
       [nan, nan, nan, nan, nan],
       [nan, nan, nan, nan, nan],
       [nan, nan, nan, nan, nan]])
>>> 
```
I think in this case we could return `b_min` and have a system warning...

thanks!

Comment (abpwrs) [reply monai/transforms/intensity/array.py:384]: Commit https://github.com/Project-MONAI/MONAI/pull/615/commits/f4437a7e3a6704598bd3d468b6be30621b8a7c76 resolves this.

New behavior is:
```python
>>> from monai.transforms import ScaleIntensityRangePercentiles
>>> import numpy as np
>>> ScaleIntensityRangePercentiles(1, 99, 0, 1)(np.ones((5, 5)))
/Users/abpwrs/projects/MONAI/monai/transforms/intensity/array.py:241: Warning: Divide by zero (a_min == a_max)
  warn("Divide by zero (a_min == a_max)", Warning)
array([[0., 0., 0., 0., 0.],
       [0., 0., 0., 0., 0.],
       [0., 0., 0., 0., 0.],
       [0., 0., 0., 0., 0.],
       [0., 0., 0., 0., 0.]])
```

**Pull request merged**